modin/core/execution/dask/implementations/pandas_on_dask/io/io.py

```diff
@@ -170,6 +170,19 @@ def to_sql(cls, qc, **kwargs):
         # since the mapping operation is non-blocking, each partition will return an empty DF
         # so at the end, the blocking operation will be this empty DF to_pandas
 
+        import sqlalchemy as sa
+
+        if not isinstance(kwargs["con"], str) and not isinstance(
+            kwargs["con"], (sa.engine.Engine, sa.engine.Connection)
+        ):
+            return BaseIO.to_sql(qc, **kwargs)
+
+        # In the case that we are given a SQLAlchemy Connection or Engine, the objects
+        # are not pickleable. We have to convert it to the URL string and connect from
+        # each of the workers.
+        if isinstance(kwargs["con"], (sa.engine.Engine, sa.engine.Connection)):
+            kwargs["con"] = repr(kwargs["con"].engine.url)
```

Comment:
using `repr()` here feels wrong, why is it needed?

Reply:
I can use `str(...)` instead.
```
(Pdb) type(kwargs["con"].engine.url)
<class 'sqlalchemy.engine.url.URL'>
```